hydra-cluster/src/Hydra/Cluster/Util.hs

```diff
@@ -66,7 +67,20 @@ chainConfigFor ::
   [Actor] ->
   ContestationPeriod ->
   IO ChainConfig
-chainConfigFor me targetDir nodeSocket hydraScriptsTxId them contestationPeriod = do
+chainConfigFor me targetDir nodeSocket txids actors cp = chainConfigFor' me targetDir (Right nodeSocket) txids actors cp defaultDepositPeriod
+
+chainConfigFor' ::
+  HasCallStack =>
+  Actor ->
+  FilePath ->
+  Either FilePath SocketPath ->
+  -- | Transaction ids at which Hydra scripts should have been published.
+  [TxId] ->
+  [Actor] ->
+  ContestationPeriod ->
+  DepositPeriod ->
+  IO ChainConfig
+chainConfigFor' me targetDir socketOrProjectPath hydraScriptsTxId them contestationPeriod depositPeriod = do
```

Comment:
why not taking chainBackendOptions as argument instead of socketOrProjectPath?

Reply:
NetworkId is always the same so we differentiate only on either socket  or blockfrost project file.